File: NumberGuessingGame.js
var readline = require("readline");
var terminal = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let correctNumber = Math.floor(Math.random() * Math.floor(100)) + 1;
var guessCount = 0;

console.log("Guess a number between 1-100");
terminal.prompt();

terminal.on("line", function (answer) {
  guessCount++;
  let guess = parseInt(answer);

  if (guess > correctNumber) {
    console.log("You guessed to high");
    terminal.prompt();
  } else if (guess < correctNumber) {
    console.log("You guessed to low");
    terminal.prompt();
  } else if (guess === correctNumber) {
    console.log("Correct, you guessed " + guessCount + " times");
    terminal.close();
  } else {
    console.log("That wasn't a number");
  }
});

File: NumberGuessingGame.cpp
#include <iostream>
#include <ctime>
int guessCount = 0;
int guess;
int num;
int main()
{
    srand(time(nullptr));
    num = rand() % 100 + 1;

    do
    {
        std::cout << "Guess a number between 1-100: ";
        std::cin >> guess;
        guessCount++;
        if (guess < num)
        {
            std::cout << "You guessed too low, try again" << std::endl;
        }
        else if (guess > num)
        {
            std::cout << "You guessed too high, try again" << std::endl;
        }
    } while (guess != num);

    std::cout << "Congrats!! You guessed correct" << std::endl;
    std::cout << "You guessed " << guessCount << " times" << std::endl;
    system("pause");
    return 0;
}